src/rulesets/LinearAlgebra/norm.jl

```diff
@@ -17,20 +17,25 @@ function frule((_, Δx), ::typeof(norm), x::Number, p::Real)
     return y, ∂y
 end
 
-function rrule(
-    ::typeof(norm),
-    x::Union{StridedArray, LinearAlgebra.AbstractTriangular, Diagonal},
-    p::Real,
-)
+function rrule(::typeof(norm), x::AbstractArray, p::Real)
```

Comment:
```suggestion
# HACK: are very broad in what we accept here, as a hack around Zygote being bad at StaticArrays
# c.f. https://github.com/FluxML/Zygote.jl/issues/860
function rrule(::typeof(norm), x::AbstractArray, p::Real)
```

Reply:
Not sure why this is a hack. If Zygote were arbitrarily good at scalar code, why would it need any array rules? But it's not. So we write array rules. 

One reason is that sometimes the array-level operation is more stable than the naiive one. That applies equally to StaticArrays and Arrays.